syftlib/src/main/java/org/openmined/syft/datasource/DataLoader.kt

```diff
@@ -0,0 +1,42 @@
+package org.openmined.syft.datasource
+
+import org.openmined.syft.datasource.Sampler
+import org.openmined.syft.datasource.RandomSampler
+import org.openmined.syft.datasource.SequentialSampler
+import org.openmined.syft.datasource.Dataset
+
+/**
+ * Data loader. Combines a dataset and a sampler, and provides an iterable over
+ * the given dataset. It supports map-style datasets with single-process loading
+ * and customizing loading order.
+ * @param dataset (Dataset)from which to load the data.
+ * @param batchSize (Int, optional): how many samples per batch to load (default: ``1``).
+ * @param shuffle (Boolean, optional): set to ``True`` to have the data reshuffled
+ *                  at every epoch (default: ``False``).
+ * @param sampler (Sampler or Iterable, optional): defines the strategy to draw
+ *                  samples from the dataset. Can be any ``Iterable`` with ``__len__``
+ *                  implemented. If specified, :attr:`shuffle` must not be specified.
+ * @param dropLast (Boolean, optional): set to ``True`` to drop the last incomplete batch,
+ *                  if the dataset size is not divisible by the batch size. If ``False`` and
+ *                  the size of dataset is not divisible by the batch size, then the last batch
+ *                  will be smaller. (default: ``False``)
+ */
+class DataLoader(var dataset: Dataset, var batchsize: Int = 1, var shuffle: Boolean = false,
```

Comment:
Either pass the `sampler` instance or `shuffle` but not both (I'd prefer the instance).

Reply:
> Either pass the `sampler` instance or `shuffle` but not both (I'd prefer the instance).

hmm, but this is supposed to allow user to choose to use RandomSampler or SequentialSampler.
Not sure how to do that without the shuffle boolean

Reply:
The user needs to pass the sampler in the constructor. He/she would have built it before passing it to this class.